meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/adapter/StateChangeAdapter.java

```diff
@@ -0,0 +1,75 @@
+package org.apache.helix.metaclient.impl.zk.adapter;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.helix.metaclient.api.ConnectStateChangeListener;
+import org.apache.helix.metaclient.exception.MetaClientTimeoutException;
+import org.apache.helix.metaclient.impl.zk.util.ZkMetaClientUtil;
+import org.apache.helix.zookeeper.zkclient.IZkStateListener;
+import org.apache.helix.zookeeper.zkclient.exception.ZkInterruptedException;
+import org.apache.zookeeper.Watcher;
+
+
+public class StateChangeAdapter implements IZkStateListener {
+  private final ConnectStateChangeListener _listener;
+
+  public StateChangeAdapter(ConnectStateChangeListener listener) {
+    _listener = listener;
+  }
+
+  @Override
+  public void handleStateChanged(Watcher.Event.KeeperState state) throws Exception {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public void handleNewSession(String sessionId) throws Exception {
+  }
```

Comment:
Should we throw `UnsupportedOperationException` for this one too?

Reply:
handleNewSession will be invoked when new connection is established. This event is covered in another callback. Will not expose this to user.